queries.sql

```diff
@@ -23,3 +23,77 @@ SELECT name FROM animals WHERE name <> 'Gabumon';
 
 -- Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+
+-- Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'unspecified';
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+-- Inside a transaction:
+-- Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
+-- Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
+-- Commit the transaction.
+-- Verify that change was made and persists after commit.
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+COMMIT;
+SELECT * FROM animals;
+
+-- Inside a transaction delete all records in the animals table, then roll back the transaction.
+-- After the rollback verify if all records in the animals table still exists.
+BEGIN TRANSACTION;
+DELETE FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+-- Inside a transaction:
+-- Delete all animals born after Jan 1st, 2022.
+-- Create a savepoint for the transaction.
+-- Update all animals' weight to be their weight multiplied by -1.
+-- Rollback to the savepoint
+-- Update all animals' weights that are negative to be their weight multiplied by -1.
+-- Commit transaction
+-- Verify that change was made and persists after commit.
+BEGIN TRANSACTION;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT update_weight;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO update_weight;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
+SELECT * FROM animals;
+
+
+-- How many animals are there?
+SELECT COUNT(*) AS total_animals FROM animals;
+
+-- How many animals have never tried to escape?
+SELECT COUNT(*) AS total_animals FROM animals WHERE escape_attempts=0;
+
+-- What is the average weight of animals?
+SELECT ROUND(AVG(weight_kg), 2) AS average_weight FROM animals;
+
+-- Who escapes the most, neutered or not neutered animals?
+SELECT
+  neutered,
+  COUNT(escape_attempts) AS number_of_escapes
+FROM animals
+WHERE escape_attempts > 0
+GROUP BY neutered
+ORDER BY number_of_escapes DESC
+
+-- What is the minimum and maximum weight of each type of animal?
+SELECT MIN(weight_kg) AS minimum_weight, MAX(weight_kg) AS maximum_weight FROM animals
+
```

Comment:
- Kindly Refactor this according to [this query](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=What%20is%20the%20minimum%20and%20maximum%20weight%20of%20each%20type%20of%20animal%3F), to display minimum weight for **each type of animal**, not for all animals at once